lib/src/main/kotlin/com/lemonappdev/konsist/core/declaration/KoDoc.kt

```diff
@@ -0,0 +1,178 @@
+package com.lemonappdev.konsist.core.declaration
+
+import org.jetbrains.kotlin.kdoc.psi.api.KDocElement
+import org.jetbrains.kotlin.kdoc.psi.impl.KDocSection
+
+class KoDoc(private val kDocElement: KDocElement) {
+    val text: String by lazy { kDocElement.text }
+
+    val description by lazy {
+        kDocElement
+            .children
+            .filterIsInstance<KDocSection>()
+            .first()
+            .text
+            .substringBefore("@")
+            .split("\n")
+            .map {
+                it
+                    .substringAfter("*")
+                    .removePrefix(" ")
+            }
+            .toMutableList()
+            .also { it.removeIf { element -> element.isBlank() } }
+            .joinToString("\n")
+    }
+
+    private val tags by lazy {
+        val firsts = kDocElement
+            .children
+            .filterIsInstance<KDocSection>()
+            .first()
+            .text
+            .replaceBefore('@', "")
+            .split("\n")
+            .map {
+                it
+                    .substringAfter("*")
+                    .removePrefix(" ")
+            }
+            .toMutableList()
+            .also { it.removeIf { element -> element.isBlank() } }
+
+        val others = kDocElement
+            .children
+            .filterIsInstance<KDocSection>()
+            .toMutableList()
+            .also { it.removeFirst() }
+            .map { it.text }
+            .flatMap {
+                it.split("\n")
+                    .map { line ->
+                        line
+                            .substringAfter("*")
+                            .removePrefix(" ")
+                    }
+                    .toMutableList()
+                    .also { line -> line.removeIf { element -> element.isBlank() } }
+            }
+
+        firsts + others
+    }
+
+    val paramBlockTags by lazy {
+        val tag = this.tags.filter { it.startsWith("@param") }
```

Comment:
Every String should be an enum. This way you could go through all of the tags...  

```kotlin
	val regex = "@(.*?)[ ]".toRegex()
    val string = """
    @AB
    @return assa sas 
    
    @constructor sas a sasa as s asa
    as sadasdasd
    
    @throws dsasddas daads
    """
    
	val matches = regex.findAll(string)
    val names = matches.map { it.value }.joinToString()
    println(names)
```

... look for tag in enum and print a warning when doc has unsupported tag.